test/library/characterization/test_stark_ramsey_xy.py

```diff
@@ -133,3 +137,151 @@ def test_stark_offset_always_positive(self):
+
+        # Check frequency shift
+        circ0_x_set_freq = circs0_x_starkv.blocks[0].frequency
+        circ0_y_set_freq = circs0_y_starkv.blocks[0].frequency
+        # This must induce negative frequency shift.
+        # This tone frequency must be greater than f01.
+        self.assertGreater(circ0_x_set_freq, f01)
+        self.assertGreater(circ0_y_set_freq, f01)
+
+        circ1_x_set_freq = circs1_x_starkv.blocks[0].frequency
+        circ1_y_set_freq = circs1_y_starkv.blocks[0].frequency
+        # This must induce positive frequency shift.
+        # This tone frequency must be lower than f01.
+        self.assertLess(circ1_x_set_freq, f01)
+        self.assertLess(circ1_y_set_freq, f01)
+
+        # Check amplitude is always positive
+        circ0_x_set_amp = circs0_x_starkv.blocks[1].pulse.parameters["amp"]
+        circ0_y_set_amp = circs0_y_starkv.blocks[1].pulse.parameters["amp"]
+        self.assertEqual(circ0_x_set_amp, 0.1)
+        self.assertEqual(circ0_y_set_amp, 0.1)
+        circ1_x_set_amp = circs1_x_starkv.blocks[1].pulse.parameters["amp"]
+        circ1_y_set_amp = circs1_y_starkv.blocks[1].pulse.parameters["amp"]
+        self.assertEqual(circ1_x_set_amp, 0.1)
+        self.assertEqual(circ1_y_set_amp, 0.1)
+
+    def test_circuit_valid_delays(self):
+        """Test Stark tone durations are valid."""
+        backend = FakeHanoiV2()
+        exp = StarkRamseyFast(
+            physical_qubits=[0],
+            backend=backend,
+            stark_amps=[0.1],
+        )
+        circs = exp.circuits()
+
+        for circ in circs:
+            stark_v = next(iter(circ.calibrations["StarkV"].values()))
+            self.assertEqual(stark_v.duration % backend.target.granularity, 0)
+            stark_u = next(iter(circ.calibrations["StarkU"].values()))
+            self.assertEqual(stark_u.duration % backend.target.granularity, 0)
+
+    def test_parameter_rounding(self):
+        """Test removing unnecessary parameter value precision."""
+        backend = FakeHanoiV2()
+        exp = StarkRamseyFast(
+            physical_qubits=[0],
+            backend=backend,
+            stark_amps=[0.12345678910111213],
+        )
+        params = exp.parameters()
+        self.assertEqual(params[0], 0.1234568)
+
+    @named_data(
+        ["ideal_quadratic", 0.5, 0.0, 30e6, 0.0, 0.0, -30e6, 0.0, 0.0, 0.5],
+        ["with_all_terms", 0.5, 15e6, 200e6, -100e6, 15e6, -200e6, -100e6, 300e3, 0.5],
+        ["with_spam_like", 0.3, 0.0, 30e6, 0.0, 0.0, -30e6, 0.0, 0.0, 0.4],
+        ["asymmetric_shift", 0.4, -20e6, 200e6, -100e6, -15e6, -180e6, -90e6, 200e3, 0.5],
+        ["large_cubic_term", 0.5, 10e6, 15e6, 30e6, 5e6, -10e6, 40e6, 0.0, 0.5],
+        ["positive_c2n", 0.5, 0.0, 50e6, 0.0, 0.0, 50e6, 0.0, 0.0, 0.5],
+        ["negative_c2p", 0.5, 0.0, -50e6, 0.0, 0.0, -50e6, 0.0, 0.0, 0.5],
+    )
+    @unpack
+    def test_ramsey_fast_analysis(self, amp, c1p, c2p, c3p, c1n, c2n, c3n, ferr, off):
```

Comment:
Tested in https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1009#issuecomment-1614138877

Reply:
(Commenting here since it is a thread) The data for the finite frequency case look close to the real data! Not sure what is going on when the Stark frequency is set to 0 but that might be a weird case. I hope we can figure out what is causing the weird behavior.

Reply:
Actually it gradually changes the oscillation period as I change the carrier frequency.